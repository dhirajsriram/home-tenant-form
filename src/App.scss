.main-container{
    .main-content{
        display: flex;
        align-items: center;
        min-height: calc(100vh - 160px);
    }
}