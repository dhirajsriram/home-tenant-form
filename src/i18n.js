import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { initReactI18next } from "react-i18next";

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    // we init with resources
    resources: {
      de: {
        translations: {
          "Name": "Name",
          "Email":"Email",
          "Phone": "Phone",
          "Salary": "Gehalt",
          "Review": "Rezension",
          "Enter your name": "Gib deinen Namen ein",
          "First Name": "Vorname",
          "First name": "Vorname",
          "Last Name": "Nachname",
          "Last name": "Nachname",
          "Enter your Email Address": "Geben sie ihre E-Mail Adresse ein",
          "Enter your phone": "Geben Sie Ihr Telefon ein",
          "Enter your Salary": "Geben Sie Ihr Gehalt ein",
          "Kindly review your input": "Bitte überprüfen Sie Ihre Eingabe",
          "Over 4000": "Mehs als 4000",
          "submit":"Registrieren",
          next: "weiter",
          back: "Bisherige"
        },
      },
      en: {
        translations: {
          "Name": "Name",
          "Email":"Email",
          "Phone": "Phone",
          "Salary": "Salary",
          "Review": "Review", 
          "Enter your name": "Enter your name",
          "First Name": "First Name",
          "First name": "First Name",
          "Last Name": "Last Name",
          "Last name": "Last Name",
          "Enter your Email Address": "Enter your Email Address",
          "Enter your phone": "Enter your Email Phone",
          "Enter your Salary": "Enter your Email Salary",
          "Kindly review your input":"Kindly review your input",
          "Over 4000": "Over 4000",
          "submit":"submit",
          next: "next",
          back: "back"
        },
      },
    },
    fallbackLng: "en",
    debug: true,

    // have a common namespace used around the full app
    ns: ["translations"],
    defaultNS: "translations",

    keySeparator: false, // we use content as keys

    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
