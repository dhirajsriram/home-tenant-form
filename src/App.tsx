import React, { Component } from "react";
import { Route, Switch } from "react-router-dom";
import Loader from "./common/loader/loader";
import routes from "./routes";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import { grey } from "@material-ui/core/colors";
import "./App.scss";
import { Trans } from "react-i18next";

const Header = React.lazy(() => import("./common/header/header"));
const Stepper = React.lazy(() => import("./common/stepper/Stepper"));
const theme = createMuiTheme({
  palette: {
    primary: {
      main: grey[800],
    },
    secondary: {
      main: grey[800],
    },
  },
});

class App extends Component {
  render() {
    return (
      <React.Suspense fallback={<Loader />}>
        <MuiThemeProvider theme={theme}>
          <Trans>
            <div className="main-container">
              <Header />
              <div className="main-content">
                <Stepper />
                <Switch>
                  {routes.map((route: any, index: number) => (
                    <Route
                      key={route.path}
                      path={route.path}
                      render={(props: any) => <route.component {...props} />}
                    />
                  ))}
                </Switch>
              </div>
            </div>
          </Trans>
        </MuiThemeProvider>
      </React.Suspense>
    );
  }
}

export default App;
