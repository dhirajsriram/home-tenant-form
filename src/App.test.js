import React from "react";
import App from "./App";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import { render, wait, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import allReducers from "./store/reducer";
import { createStore } from "redux";
import "./i18n";

let wrapper;
configure({ adapter: new Adapter() });

beforeEach(() => {
  wrapper = shallow(<App />);
});

afterEach(cleanup);

const store = createStore(allReducers);

describe("<App>", () => {
  it("Unknown routes should be navigated to a 404 page", async () => {
    const fourzerofour = "404!";
    const { getByText } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/sddsdweq"]}>
          <App />
        </MemoryRouter>
      </Provider>
    );
    await wait(() => expect(getByText(fourzerofour)).toBeDefined());
  });
  it("Default routes to be navigated to first step of the form", async () => {
    const hometext = "Name";
    const { getByText } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]}>
          <App />
        </MemoryRouter>
      </Provider>
    );
    await wait(() => expect(getByText(hometext)).toBeDefined());
  });
  it("Default routes to be navigated to first step of the form", async () => {
    expect(wrapper).toMatchSnapshot();
  });
});



