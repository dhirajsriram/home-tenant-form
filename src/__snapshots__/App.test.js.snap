// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> Default routes to be navigated to first step of the form 1`] = `ShallowWrapper {}`;
